pkgs/applications/version-management/git-and-tools/git/ssh-path.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/connect.c b/connect.c
+index fd7ffe1..20cd992 100644
+--- a/connect.c
++++ b/connect.c
+@@ -768,7 +768,7 @@ struct child_process *git_connect(int fd[2], const char *url,
+ 
+ 				ssh = getenv("GIT_SSH");
+ 				if (!ssh)
+-					ssh = "ssh";
++					ssh = "@ssh@";
+ 
+ 				ssh_dup = xstrdup(ssh);
+ 				base = basename(ssh_dup);
+diff --git a/git-gui/lib/remote_add.tcl b/git-gui/lib/remote_add.tcl
+index 50029d0..2059b26 100644
+--- a/git-gui/lib/remote_add.tcl
++++ b/git-gui/lib/remote_add.tcl
+@@ -141,7 +141,7 @@ method _add {} {
+ 		     || [regexp {([^:][^:]+):(.+)} $location xx host path]} {
+ 			set ssh ssh
+ 			if {[info exists env(GIT_SSH)]} {
+-				set ssh $env(GIT_SSH)
++				set @ssh@ $env(GIT_SSH)
```

Comment:
This change is incorrect, it will expand to:
```tcl
set /nix/store/... $env(GIT_SSH)
```
which sets a variable named "/nix/store/..." to the string "ssh" (not the desired result).

You just need to change the 'set ssh' line to:

```tcl
set ssh "@ssh@"
```

and leave the if ... block intact so the GIT_SSH env var can override the hardcoded location.